Zend/Optimizer/sccp.c

```diff
@@ -83,10 +83,10 @@ typedef struct _sccp_ctx {
 	zval bot;
 } sccp_ctx;
 
-#define TOP ((zend_uchar)-1)
-#define BOT ((zend_uchar)-2)
-#define PARTIAL_ARRAY ((zend_uchar)-3)
-#define PARTIAL_OBJECT ((zend_uchar)-4)
+#define TOP ((uint_least8_t)-1)
+#define BOT ((uint_least8_t)-2)
+#define PARTIAL_ARRAY ((uint_least8_t)-3)
+#define PARTIAL_OBJECT ((uint_least8_t)-4)
```

Comment:
The values would differ on some hypothetical arch that does not have an 8 bit integer.

Reply:
True, but these different values would be correct on that arch, wouldn't they?
(Isn't this cast UB anyway?)